samples/drawing/drawing.cpp

```diff
@@ -2055,13 +2055,17 @@ void MyCanvas::OnMouseMove(wxMouseEvent &event)
 
         wxDCOverlay overlaydc( m_overlay, &dc );
         overlaydc.Clear();
-#ifdef __WXMAC__
-        dc.SetPen( *wxGREY_PEN );
-        dc.SetBrush( wxColour( 192,192,192,64 ) );
-#else
-        dc.SetPen( wxPen( *wxLIGHT_GREY, 2 ) );
-        dc.SetBrush( *wxTRANSPARENT_BRUSH );
-#endif
+        if ( m_overlay.IsNative() )
+        {
+            dc.SetPen( *wxGREY_PEN );
+            dc.SetBrush( wxColour( 192,192,192,64 ) );
```

Comment:
Does this really work under MSW? I think only ports using wxGC for implementing wxDC (i.e. Mac and GTK 3) support alpha channel in brush colours.

Reply:
This is true. But for `wxMSW` the alpha channel is just ignored and this is not a problem because the overlay transparency is controlled by this call:
`::SetLayeredWindowAttributes(hwnd, 0, 128, LWA_COLORKEY|LWA_ALPHA)`

BTW if I change **128** to **64** and set the pen to `dc.SetPen( wxColour(20, 20, 20 );` the rubber band looks almost like the native one!

Reply:
I think the current code is better because it's not misleading: people looking at the sample could believe that using alpha works under MSW when, as you say, it's just ignored there.